src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -545,27 +551,20 @@ bool isValidScenario
 
         /// <summary>
         /// Test method to check that an exception is thrown when LinkingObject is null
-        /// and either of SourceFields and TargetFields is null in the config.
+        /// and SourceFields and TargetFields are null in the config.
         /// And the foreignKey pair between source and target is not defined in the database as well.
         /// Also verify that after adding foreignKeyPair in the Database, no exception is thrown.
         /// </summary>
-        [DataRow(null, new string[] { "das" }, null, DisplayName = "SourceFields is null")]
-        [DataRow(new string[] { "id" }, null, null, DisplayName = "TargetFields is null")]
```

Comment:
these tests need to be added back. Their presence for this specific test implies that a mismatch in sourcefield and target field count is fine as long as a FK is defined in the database which would ultimately provide the missing metadata to dab.

Reply:
This config is valid, test in your environment:

```json
{
  "$schema": "https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json",
  "data-source": {
    "database-type": "mssql",
    "connection-string": "<CONN_STRING>",
    "options": {
      "set-session-context": true
    }
  },
  "runtime": {
    "rest": {
      "enabled": true,
      "path": "/api",
      "request-body-strict": true
    },
    "graphql": {
      "enabled": true,
      "path": "/graphql",
      "allow-introspection": true,
      "multiple-mutations": {
        "create": {
          "enabled": true
        }
      }
    },
    "host": {
      "cors": {
        "origins": [
          "http://localhost:5000"
        ],
        "allow-credentials": false
      },
      "authentication": {
        "provider": "Simulator"
      },
      "mode": "development"
    }
  },
  "entities": {
    "Publisher": {
      "source": {
        "object": "publishers",
        "type": "table"
      },
      "graphql": {
        "enabled": true,
        "type": {
          "singular": "Publisher",
          "plural": "Publishers"
        }
      },
      "rest": {
        "enabled": true
      },
      "permissions": [
        {
          "role": "anonymous",
          "actions": [
            "*"
          ]
        }
      ],
      "relationships": {
        "books": {
          "cardinality": "many",
          "target.entity": "Book",
          "source.fields": ["id"],
          "target.fields": [],
          "linking.source.fields": [],
          "linking.target.fields": []
        }
      }
    },
    "Book": {
      "source": {
        "object": "books",
        "type": "table"
      },
      "graphql": {
        "enabled": true,
        "type": {
          "singular": "book",
          "plural": "books"
        }
      },
      "rest": {
        "enabled": true
      },
      "permissions": [
        {
          "role": "anonymous",
          "actions": [
            "*"
          ]
        }
      ],
      "mappings": {
        "id": "id",
        "title": "title"
      },
      "relationships": {
        "publishers": {
          "cardinality": "one",
          "target.entity": "Publisher",
          "source.fields": [],
          "target.fields": [],
          "linking.source.fields": [],
          "linking.target.fields": []
        }
      }
    }
  }
}

```

note: `publisher.relationship.books` has the following contents for source/target fields are mismatched count:
```json
      "relationships": {
        "books": {
          "cardinality": "many",
          "target.entity": "Book",
          "source.fields": ["id"],
          "target.fields": [],
          "linking.source.fields": [],
          "linking.target.fields": []
        }
```

The DB has a FK for this relationship, which dab resolves and then allows this config because the db metadata makes up for the missing `target.fields` value.

Reply:
intent is to use config target/source only and if not defined use db metadata. DB metadata will not fill in the blanks in config where fields are erroneously left out.